workflow/executor/executor.go

```diff
@@ -238,6 +243,130 @@ func (we *WorkflowExecutor) LoadArtifacts(ctx context.Context) error {
 	return nil
 }
 
+func (we *WorkflowExecutor) ProcessData(ctx context.Context) error {
```

Comment:
I think historically executor func names map directly to the command name, so this should. be called `Data`?

Reply:
maybe create a file name `executor_data.go` with all data funcs, so it is easy to associate the tests in `executor_data_test.go`